core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -393,6 +400,29 @@ public static String getCacheDirective(final Message msg) {
         return getApplicationProperty(msg.getApplicationProperties(), APP_PROPERTY_CACHE_CONTROL, String.class);
     }
 
+    /**
+     * Adds the <em>time until disconnect</em> property to an AMQP 1.0 message.
+     * <p>
+     * The value is put to the message's <em>application-properties</em> under key
+     * {@link #APP_PROPERTY_DEVICE_TTD}.
+     *
+     * @param msg The message to add the property to.
+     * @param timeUntilDisconnect The value of the property.
+     */
+    public static void addTimeUntilDisconnect(final Message msg, final Integer timeUntilDisconnect) {
```

Comment:
what if ttd is `null`

Reply:
Right, javadoc for NPE was missing, thx!

Reply:
why is this an `Integer` instead of an `int` anyway if setting it to `null` is not allowed? Is it allowed to set it to a negative value or to 0? this would probably require adding some checks and throwing an `IllegalArgumentException` instead of a `NullPointerException` then ...